WordPress/Classes/ViewRelated/Reader/ReaderMenuViewModel.swift

```diff
@@ -0,0 +1,615 @@
+
+        // Rebuild!
+        setupDefaultsSection()
+
+        if isSignedInWPCom() {
+            setupListsSection()
+        }
+
+        setupTagsSection()
+    }
+
+
+    // MARK: - Default Section
+
+
+    /// Sets up the default fetched results controller.
+    ///
+    func setupDefaultResultsController() {
+        let fetchRequest = NSFetchRequest(entityName: "ReaderDefaultTopic")
+        let sortDescriptor = NSSortDescriptor(key: "title", ascending: true, selector: #selector(NSString.localizedCaseInsensitiveCompare))
+        fetchRequest.sortDescriptors = [sortDescriptor]
+
+        defaultsFetchedResultsController = NSFetchedResultsController(fetchRequest: fetchRequest,
+                                                                      managedObjectContext: ContextManager.sharedInstance().mainContext,
+                                                                      sectionNameKeyPath: nil,
+                                                                      cacheName: nil)
+        defaultsFetchedResultsController.delegate = self
+
+        do {
+            let _ = try defaultsFetchedResultsController.performFetch()
+        } catch {
+            DDLogSwift.logError("There was a problem fetching default topics for the menu.")
+            assertionFailure("There was a problem fetching default topics.")
+        }
+    }
+
+
+    /// Sets up the defaults section and its corresponding NSFetchedResultsController.
+    ///
+    func setupDefaultsSection() {
+        let section = ReaderMenuSection(title: NSLocalizedString("General", comment:"Section title of the default reader items."), type: .Defaults)
+        sections.append(section)
+
+        buildDefaultSectionItems()
+    }
+
+
+    /// Builds (or rebuilds) the items for the default menu section.
+    /// Since the default section shows items representing things other than topics,
+    /// we construct and cache menu items in an array with the desired item order.
+    ///
+    func buildDefaultSectionItems() {
+        defaultSectionItems.removeAll()
+
+        // Create menu items from the fetched results
+        if let fetchedObjects = defaultsFetchedResultsController.fetchedObjects {
+            for topic in fetchedObjects {
+                guard let abstractTopic = topic as? ReaderAbstractTopic else {
+                    continue
+                }
+
+                var item = ReaderMenuItem(title: topic.title, type: .Topic, icon: nil, topic: abstractTopic)
+                if ReaderHelpers.topicIsFollowing(abstractTopic) {
+                    item.order = ReaderDefaultMenuItemOrder.Followed.rawValue
+                    item.icon = Gridicon.iconOfType(.CheckmarkCircle)
+                } else if ReaderHelpers.topicIsDiscover(abstractTopic) {
+                    item.order = ReaderDefaultMenuItemOrder.Discover.rawValue
+                    item.icon = Gridicon.iconOfType(.MySites)
+                } else if ReaderHelpers.topicIsLiked(abstractTopic) {
```

Comment:
Just wondering if those `ReaderHelpers` would fit as an extension to `ReaderAbstractTopic` (or so)